### PR TITLE
swarm API

### PR DESCRIPTION
**Pull request by diasdavid**
> **WIP**

- API calls
  - [x] ipfs.swarm.addrs
  - [x] ipfs.swarm.connect
  - [x] ipfs.swarm.disconnect
  - [x] ipfs.swarm.filters
  - [x] ipfs.swarm.peers
- [ ] Works in js-ipfs
- [x] Works in js-ipfs-api
- requires:
  - [x] https://github.com/ipfs/interface-ipfs-core/issues/53

### CODE REVIEW

**Comment by diasdavid** [comment]
@dignifiedquire please review this one when you can :)

File: API/swarm/README.md [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+##### `JavaScript` - ipfs.swarm.peers([callback])
+
+`callback` must follow `function (err, peerInfos) {}` signature, where `err` is an error if the operation was not successful. `peerInfos` will be an array of [PeerInfo]().
+
+If no `callback` is passed, a promise is returned.
+
+Example:
+
+```JavaScript
+ipfs.swarm.peers(function (https://twitter.com/hichaelmart/status/763777073265709056err, peerInfos) {})
```
Comment (dignifiedquire) [inline API/swarm/README.md]: Why is there a twitter status in here?

File: API/swarm/README.md [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+```
+
+------------------------------
+> NOT IMPLEMENTED YET
+
+#### `filters`
```
Comment (dignifiedquire) [inline API/swarm/README.md]: It would be nice to have a little bit more explanation of what `filters` actually do

File: package.json [hunk 3]
```diff
@@ -2,7 +2,7 @@
   "name": "interface-ipfs-core",
   "version": "0.11.0",
   "description": "A test suite and interface you can use to implement a IPFS core interface.",
-  "main": "lib/index.js",
+  "main": "src/index.js",
```
Comment (dignifiedquire) [inline package.json]: TODO: remove before merge

File: src/swarm.js [hunk 4]
```diff
@@ -0,0 +1,123 @@
+/* eslint-env mocha */
+/* eslint max-nested-callbacks: ["error", 8] */
+
+'use strict'
+
+const expect = require('chai').expect
+const series = require('run-series')
+
+module.exports = (common) => {
+  describe.only('.swarm', () => {
```
Comment (dignifiedquire) [inline src/swarm.js]: TODO: remove before merge

File: API/swarm/README.md [hunk 5]
```diff
@@ -0,0 +1,143 @@
+swarm API
```
Comment (dignifiedquire) [inline API/swarm/README.md]: Capitalize?

File: API/swarm/README.md (hunk 1 above)
Comment (diasdavid) [inline API/swarm/README.md]: 😂

**Comment by diasdavid** [comment]
Thank you for the CR :)

**Pull request merged**